### PR TITLE
Halo update switch

### PR DESCRIPTION
**Pull request by twicki**
## Purpose
- Enable MPI-distributed compilation by allowing halo exchanges to be switched off in parallel tests.
- This is a workaround for the problem of compiler crashes (`gcc` or `nvcc`) that occur when halo exchanges are enabled during compilation.

## Code changes:

- Update `run_on_daint.sh` to compile on multiple nodes
-  Add `--disable_halo_exchange` option to `dynamics`
- Wrap `if` statements around halo update code in parallel tests
- Add `do_halo_exchange` getter and setter to `global_config`

## Infrastructure changes:

- Enabled gtcuda c128 runs both on caching and performance nightly Jenkins builds

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

Co-authored-by: Eddie Davis <EddieD@vulcan.com>
Co-authored-by: Rhea George <RheaG@vulcan.com>

### CODE REVIEW

**Review by rheacangeo — APPROVED** [review]
lgtm!

File: examples/standalone/benchmarks/run_on_daint.sh [hunk 1]
```diff
@@ -89,17 +89,24 @@ else
 fi
 
 
+split_path=(${data_path//\// })
+experiment=${split_path[-1]}
 
+# if [ ! -d $(pwd)/.gt_cache_000000 ]; then
```
Comment (eddie-c-davis) [inline examples/standalone/benchmarks/run_on_daint.sh:95]: Do we want to fix this to use the results of the caching build?

Comment (rheacangeo) [reply examples/standalone/benchmarks/run_on_daint.sh:95]: good point we probably should add this back in conjunction with changing the name of the cache plan

File: fv3core/utils/gt4py_utils.py [hunk 2]
```diff
@@ -52,6 +52,7 @@
 # Logger instance
 logger = logging.getLogger("fv3ser")
 
+
```
Comment (rheacangeo) [inline fv3core/utils/gt4py_utils.py:55]: the linter let us add extra whitespace?! I am shocked.

File: fv3core/utils/gt4py_utils.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 logger = logging.getLogger("fv3ser")
 
 # Set to "False" to skip validating gt4py stencil arguments
-validate_args = True
+validate_args = False
```
Comment (rheacangeo) [inline fv3core/utils/gt4py_utils.py:26]: We may want to switch this back

Comment (eddie-c-davis) [inline fv3core/utils/gt4py_utils.py:26]: We should probably set this back to `True`...

**Pull request merged**